### ISSUE TITLE
A question about _optimizer_step size of [original_param_groups[group_no]]

### ISSUE THREAD

**Issue by Ind1x1** [microsoft/DeepSpeed#6624]
def _optimizer_step(self, group_no, id):
        original_param_groups = self.optimizer.param_groups
        self.optimizer.param_groups = [original_param_groups[group_no]]
        for gpu_param in self.optimizer.param_groups[0]['params']:
            print(f"Param: GPU {id} Param Size = {gpu_param.size()}")
        # Disabling this as the C++ side copy & synchronize is not working correctly
        #from deepspeed.ops.adam import DeepSpeedCPUAdam
        #if type(self.optimizer) == DeepSpeedCPUAdam and self.dtype == torch.half:
        #    self.optimizer.step(fp16_param_groups=[self.get_bit16_param_group(group_no)])
        #else:
        #    self.optimizer.step()
        self.optimizer.step()
        self.optimizer.param_groups = original_param_groups

**when i try to print the size of [original_param_groups[group_no]], I noticed that is size are fluctuating. May I ask why this is so?  **

`Param: GPU 2tensor([ 0.0098,  0.0019,  0.0012,  ...,  0.0142, -0.0315, -0.0107],
       device='cuda:2', requires_grad=True)
Param: GPU 4 Param Size = torch.Size([33280])
Param: GPU 5 Param Size = torch.Size([33280])
Param: GPU 1 Param Size = torch.Size([33280])
Param: GPU 6 Param Size = torch.Size([33280])
Param: GPU 3 Param Size = torch.Size([33280])
Param: GPU 0 Param Size = torch.Size([33280])
Param: GPU 7 Param Size = torch.Size([33280])
Param: GPU 2 Param Size = torch.Size([33280])
Param: GPU 3tensor([0.5547, 0.5234, 0.4336,  ..., 0.5937, 0.5977, 0.5820], device='cuda:3',
       requires_grad=True)
Param: GPU 1tensor([2.6250, 2.3125, 2.6250,  ..., 0.5742, 0.5664, 0.5859], device='cuda:1',
       requires_grad=True)
Param: GPU 5tensor([1.0625, 0.5977, 0.4805,  ..., 0.5977, 0.2930, 0.5977], device='cuda:5',
       requires_grad=True)
Param: GPU 4tensor([0.5898, 0.5898, 0.5820,  ..., 0.5039, 0.4883, 0.5039], device='cuda:4',
       requires_grad=True)
Param: GPU 7tensor([0.4512, 0.7422, 0.2344,  ..., 0.5273, 0.4863, 0.4238], device='cuda:7',
       requires_grad=True)
Param: GPU 6tensor([0.5938, 0.5898, 0.5938,  ..., 0.3379, 0.3789, 0.4375], device='cuda:6',
       requires_grad=True)
Param: GPU 0tensor([0.0535, 0.2080, 0.4473,  ..., 2.5781, 2.2656, 2.7500], device='cuda:0',
       requires_grad=True)
Param: GPU 2tensor([0.5859, 0.5352, 0.5820,  ..., 0.5078, 0.5508, 0.5234], device='cuda:2',
       requires_grad=True)
{'loss': 1.1986, 'grad_norm': 18.082178584098298, 'learning_rate': 1.5789473684210526e-06, 'epoch': 0.05}                                                                                                                                                                        
  2%|███▊                                                                                                                                                                                                                                        | 3/183 [00:17<17:01,  5.67s/it]Param: GPU 5 Param Size = torch.Size([1003749376])
Param: GPU 6 Param Size = torch.Size([1003749376])Param: GPU 0 Param Size = torch.Size([1003749376])Param: GPU 3 Param Size = torch.Size([1003749376])

Param: GPU 4 Param Size = torch.Size([1003749376])
Param: GPU 1 Param Size = torch.Size([1003749376])
Param: GPU 2 Param Size = torch.Size([1003749376])Param: GPU 7 Param Size = torch.Size([1003749376])


Param: GPU 3tensor([-0.0157,  0.0014, -0.0041,  ..., -0.0101, -0.0124,  0.0164],
       device='cuda:3', requires_grad=True)Param: GPU 4tensor([ 0.0020,  0.0073,  0.0054,  ...,  0.0043,  0.0080, -0.0050],
       device='cuda:4', requires_grad=True)

Param: GPU 6tensor([ 0.0081,  0.0094,  0.0104,  ..., -0.0058, -0.0049, -0.0287],
       device='cuda:6', requires_grad=True)
Param: GPU 1tensor([-1.4771e-02, -1.0316e-02,  5.9826e-04,  ..., -1.7699e-02,
        -6.8973e-05, -8.3626e-03], device='cuda:1', requires_grad=True)
Param: GPU 5tensor([0.0099, 0.0112, 0.0082,  ..., 0.0160, 0.0112, 0.0006], device='cuda:5',
       requires_grad=True)
Param: GPU 7tensor([ 0.0068, -0.0073,  0.0077,  ..., -0.0045, -0.0031, -0.0093],
       device='cuda:7', requires_grad=True)
Param: GPU 0tensor([ 0.0013,  0.0054, -0.0022,  ...,  0.0106, -0.0018, -0.0208],
       device='cuda:0', requires_grad=True)
Param: GPU 2tensor([ 0.0098,  0.0019,  0.0012,  ...,  0.0142, -0.0315, -0.0107],
       device='cuda:2', requires_grad=True)
Param: GPU 5 Param Size = torch.Size([33280])
Param: GPU 6 Param Size = torch.Size([33280])
Param: GPU 3 Param Size = torch.Size([33280])
Param: GPU 4 Param Size = torch.Size([33280])
Param: GPU 1 Param Size = torch.Size([33280])
Param: GPU 0 Param Size = torch.Size([33280])
Param: GPU 7 Param Size = torch.Size([33280])
Param: GPU 2 Param Size = torch.Size([33280])`
